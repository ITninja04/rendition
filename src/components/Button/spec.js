/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import sinon from 'sinon'
import { Button, Provider } from '../../../dist'

test('Button renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Button>Click me</Button>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Styled Button renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Button m={2}>Click me</Button>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Styled Anchor Button renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Button href={'#'} m={2}>
        Click me
      </Button>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('onClick is called after confirmation', () => {
  const handleClick = sinon.spy()

  const component = mount(
    <Provider>
      <Button confirmation={'Sure?'} onClick={handleClick} m={2}>
        Click me
      </Button>
    </Provider>
  )

  // Popup should not be visible until button is clicked
  expect(component.find('Popover Box[children="Sure?"]')).toHaveLength(0)

  // Open the confirmation
  component.find('button[children="Click me"]').simulate('click')

  // Popup should become visible
  expect(component.find('Popover Box[children="Sure?"]')).toHaveLength(1)

  // Press Cancel
  component.find('button[children="Cancel"]').simulate('click')

  // onClick should not be called
  expect(handleClick.callCount).toEqual(0)

  // Open the confirmation
  component.find('button[children="Click me"]').simulate('click')

  // Press OK
  component.find('button[children="OK"]').simulate('click')

  // onClick should be called with an event
  expect(handleClick.callCount).toEqual(1)
  expect(handleClick.lastCall.args[0]).toHaveProperty('target')
})
