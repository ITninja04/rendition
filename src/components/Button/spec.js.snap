// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 1px solid #2A506F;
  border-radius: 20px;
  color: #444444;
  padding: 4px 30px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c2:hover {
  box-shadow: 0px 0px 0px 2px #2A506F;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  font-weight: 600;
  font-size: 14px;
  height: 38px;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 svg {
  color: inherit !important;
  font-size: 0.925em;
}

.c3:disabled {
  cursor: not-allowed;
}

.c3 {
  color: #2A506F;
  border-color: #2A506F;
}

.c3 svg {
  color: #527699 !important;
}

.c3:hover:enabled,
.c3:focus:enabled,
.c3:active:enabled {
  box-shadow: none;
  background: #2A506F;
  border-color: #2A506F;
  color: white;
  opacity: initial;
}

.c3:hover:enabled svg,
.c3:focus:enabled svg,
.c3:active:enabled svg {
  color: white !important;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <button
    className="c2 c3 "
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Styled Anchor Button renders correctly 1`] = `
.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 1px solid #2A506F;
  border-radius: 20px;
  color: #444444;
  padding: 4px 30px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c2:hover {
  box-shadow: 0px 0px 0px 2px #2A506F;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  font-weight: 600;
  font-size: 14px;
  height: 38px;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 svg {
  color: inherit !important;
  font-size: 0.925em;
}

.c3:disabled {
  cursor: not-allowed;
}

.c3 {
  color: #2A506F;
  border-color: #2A506F;
}

.c3 svg {
  color: #527699 !important;
}

.c3:hover:enabled,
.c3:focus:enabled,
.c3:active:enabled {
  box-shadow: none;
  background: #2A506F;
  border-color: #2A506F;
  color: white;
  opacity: initial;
}

.c3:hover:enabled svg,
.c3:focus:enabled svg,
.c3:active:enabled svg {
  color: white !important;
}

.c4 {
  margin: 8px;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <a
    className="c2 c3 c4"
    href="#"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    Click me
  </a>
</div>
`;

exports[`Styled Button renders correctly 1`] = `
.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 1px solid #2A506F;
  border-radius: 20px;
  color: #444444;
  padding: 4px 30px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c2:hover {
  box-shadow: 0px 0px 0px 2px #2A506F;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #00AEEF;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  font-weight: 600;
  font-size: 14px;
  height: 38px;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 svg {
  color: inherit !important;
  font-size: 0.925em;
}

.c3:disabled {
  cursor: not-allowed;
}

.c3 {
  color: #2A506F;
  border-color: #2A506F;
}

.c3 svg {
  color: #527699 !important;
}

.c3:hover:enabled,
.c3:focus:enabled,
.c3:active:enabled {
  box-shadow: none;
  background: #2A506F;
  border-color: #2A506F;
  color: white;
  opacity: initial;
}

.c3:hover:enabled svg,
.c3:focus:enabled svg,
.c3:active:enabled svg {
  color: white !important;
}

.c4 {
  margin: 8px;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <button
    className="c2 c3 c4"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;
